unblob/finder.py

```diff
@@ -82,9 +82,15 @@ def _hyperscan_match(
     if real_offset < 0:
         return _HyperscanScan.Continue
 
+    # Do not overlap with chunks of the same handler.
+    # This is mostly valid for containers, so should be configurable for each handler.
     # Skip chunk calculation if this would start inside another one,
     # similar to remove_inner_chunks, but before we even begin calculating.
-    if any(chunk.contains_offset(real_offset) for chunk in context.all_chunks):
+    if any(
+        chunk.contains_offset(real_offset)
+        for chunk in context.all_chunks
+        if chunk.handler == handler
```

Comment:
Future readers: outside of the ZIP handler, the most important change of this MR is here.